src/main/kotlin/backend/config/Config.kt

```diff
@@ -140,271 +135,6 @@ data class Config(
   }
 }
 
-abstract class ConfigFieldInfo<T> {
-  abstract val label: String
-  abstract val description: String
-  abstract val errorMessage: String
-
-  abstract fun isValid(it: String): Boolean
-  fun validate(it: String) = require(isValid(it)) { errorMessage }
-}
-
-sealed class ConfigField<out T : Any>(
-  val value: T,
-) {
```

Comment:
?

Reply:
it's weird. your refectoring ==  classes changes their placement?. leave abstract objects here, and extract custom fields to new files